deeplenstronomy/check.py

```diff
@@ -400,8 +401,13 @@ def check_input_distributions(self):
                                     errs.append("DISTRIBUTIONS." + userdist + ".STEP must be a positive integer")
                         # if params are specified, they must be in a list
                         if 'PARAMS' in self.config["DISTRIBUTIONS"][userdist].keys():
-                            if not isinstance(self.config["DISTRIBUTIONS"][userdist]['PARAMS'], list):
+                            if not isinstance(self.config["DISTRIBUTIONS"][userdist]['PARAMS']:
                                 errs.append("DISTRIBUTIONS." + userdist + ".PARAMS must be a list")
+                            else:
+                                user_df = self.config.["DISTRIBUTIONS"][userdist]['FILENAME']
+                            if len(params) != parameter_length:
```

Comment:
Note that `parameter_length` includes the `'WEIGHT'` column in the count of the number of parameters. However, in your previous checking of the number of parameters, you did not include the weight column in the count. Make sure to account for that here.

Reply:
good catch, I'll make sure it works

Reply:
What was the verdict here? Does it work correctly as you have it now or do you need to account for 'WEIGHT' in the parameter count?

Reply:
It works - the original code had subtracted the weight column in order to match some existing code in utils.py, but I went ahead and compared the lengths directly since there’s already a weight-related check in check.py

Reply:
Great thanks for looking into it!